tests/Unit/Evolution/Systems/CurvedScalarWave/Test_Characteristics.cpp

```diff
@@ -229,34 +233,137 @@ SPECTRE_TEST_CASE("Unit.Evolution.Systems.CurvedScalarWave.Characteristics",
+}
+
+template <size_t SpatialDim>
+void check_max_char_speed_compute_tag(
+    const DataVector& used_for_size) noexcept {
+  MAKE_GENERATOR(generator);
+  std::uniform_real_distribution<> distribution(0.0, 1.0);
+
+  const auto nn_generator = make_not_null(&generator);
+  const auto nn_distribution = make_not_null(&distribution);
 
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<1>::apply(
-            {{DataVector{1., -4., 3.4, 2., 5.},
-              DataVector{22., -8., 190., 6., 4.},
-              DataVector{1., -7., 31., 2., 5.},
-              DataVector{7., 8.9, 4., 2., 1.}}}) == 190.);
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<1>::apply(
-            {{DataVector{1., 4., 3., 2., 5.}, DataVector{2., 8., -10., 6., 4.},
-              DataVector{1., 7., 3., -11., 5.},
-              DataVector{7., 3., 4., 2., 1.}}}) == 11.);
-
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<2>::apply(
-            {{DataVector{1., -4., 3.4, 2., 5.},
-              DataVector{22., -8., 190., 6., 4.},
-              DataVector{1., -7., 31., 2., 5.},
-              DataVector{7., 8.9, 4., 2., 1.}}}) == 190.);
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<2>::apply(
-            {{DataVector{1., 4., 3., 2., 5.}, DataVector{2., 8., -10., 6., 4.},
-              DataVector{1., 7., 3., -11., 5.},
-              DataVector{7., 3., 4., 2., 1.}}}) == 11.);
-
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<3>::apply(
-            {{DataVector{1., -4., 3.4, 2., 5.},
-              DataVector{22., -8., 190., 6., 4.},
-              DataVector{1., -7., 31., 2., 5.},
-              DataVector{7., 8.9, 4., 2., 1.}}}) == 190.);
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<3>::apply(
-            {{DataVector{1., 4., 3., 2., 5.}, DataVector{2., 8., -10., 6., 4.},
-              DataVector{1., 7., 3., -11., 5.},
-              DataVector{7., 3., 4., 2., 1.}}}) == 11.);
+  // Randomized tensors
+  const auto gamma_1 = make_with_random_values<Scalar<DataVector>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto lapse = make_with_random_values<Scalar<DataVector>>(
+      nn_generator, nn_distribution, used_for_size);
+  const auto shift =
+      make_with_random_values<tnsr::I<DataVector, SpatialDim, Frame::Inertial>>(
+          nn_generator, nn_distribution, used_for_size);
+  const auto spatial_metric = make_with_random_values<
+      tnsr::ii<DataVector, SpatialDim, Frame::Inertial>>(
+      nn_generator, nn_distribution, used_for_size);
+
+  // Insert into databox
+  const auto box = db::create<
+      db::AddSimpleTags<
+          CurvedScalarWave::Tags::ConstraintGamma1, gr::Tags::Lapse<DataVector>,
+          gr::Tags::Shift<SpatialDim, Frame::Inertial, DataVector>,
+          gr::Tags::SpatialMetric<SpatialDim, Frame::Inertial, DataVector>>,
+      db::AddComputeTags<>>(gamma_1, lapse, shift, spatial_metric);
```

Comment:
just add the compute tag here in the `db::AddComputeTags<>` list :) then you can `get` the result out of the box.

Reply:
Can't do that as this tag doesn't inherit from `db::ComputeTag` either in this or any other evolution system where its used.

Reply:
Ok then I'm not quite sure what it even is, perhaps an interface compute tag? It should probably still inherit from `db::ComputeTag`... Anyway, no reason to hold up this PR.

Reply:
This also isn't an interface compute tag. I'm not sure exactly _what_ it is, but it's not really a tag. However, I don't know what the other systems do for implementing this. I think this is used more as a `db::apply`-ish (if that's even a thing) object. Basically, "Grab these things from the DataBox and I'll give you back the max char speed". It's more like a non-mutating mutator than it is like a tag.

Reply:
`ComputeLargestCharacteristicSpeed` is used by a  `db::apply` by `StepChoosers::Cfl` and therefore is not a compute tag.  The commit messages (and PR title) should be updated (e.g. "Fix the compute tag" -> "Fix the function used").

Reply:
Thanks for tracking that down! (not for this PR of course:) We really need to document the interfaces for these things better, because reading the code gives no indication what interface it actually implements and what the requirements on this interface are. What's missing is something that points back to something related to `StepChooser::Cfl`, or `db::apply`, or whichever code defines the interface. This is the problem I intended to address with protocols, which would add a `: tt::ConformsTo<LargestCharSpeedsComputer>` base class that points to the documentation of the interface.